team-validator.js

```diff
@@ -188,33 +188,45 @@ class Validator {
 			problems.push(`${set.species} has an invalid happiness.`);
 		}
 
-		let banReason = ruleTable.check(template.id, setHas) || ruleTable.check(template.id + 'base', setHas);
-		if (banReason) {
-			return [`${set.species} is ${banReason}.`];
-		} else {
-			banReason = ruleTable.check(toId(template.baseSpecies), setHas);
+		let templates = [template];
+		if (item.megaEvolves === template.species && !ruleTable.has('+' + item.id)) templates.push(dex.getTemplate(item.megaStone));
+		templates.forEach((template, index) => {
+			if (ruleTable.has('-mega') && template.forme in {'Mega': 1, 'Mega-X': 1, 'Mega-Y': 1}) {
+				problems.push(`Mega evolutions are banned.`);
+			}
+			let banReason = ruleTable.check(template.id, setHas) || ruleTable.check(template.id + 'base', setHas);
 			if (banReason) {
-				return [`${template.baseSpecies} is ${banReason}.`];
+				problems.push(`${template.species} is ${banReason}.`);
+			} else {
+				banReason = ruleTable.check(toId(template.baseSpecies), setHas);
+				if (banReason) {
+					problems.push(`${template.baseSpecies} is ${banReason}.`);
+				}
 			}
-		}
+			if (ruleTable.has('-unreleased') && template.isUnreleased) {
```

Comment:
It didn't get deleted, it just got moved here.

Reply:
Ohh, I saw that part but I thought it was a conditional rather than a loop through templates.

...I still don't know if a loop through templates is the right idea.